.gitignore

```diff
@@ -0,0 +1,3 @@
+.idea
```

Comment:
Can you please remove `.idea` from gitignore? The best practice is to add the IDE specific files/dirs to your machine gitignore and not to the project. Each developer can use different IDEs so the gitignore would be clutter if everyone adds theirs.

You can also check this article: https://blog.martinhujer.cz/dont-put-idea-vscode-directories-to-projects-gitignore/

Reply:
While I understand the concept, I think Rust only has a few IDEs. The trade-off is a few lines in gitignore vs having `git status` always report untracked files which is very annoying during development.

Adding stuff locally to `.gitignore` makes `git status` report that as changed. Then you'd have to add `.gitignore` itself to `.gitignore` to get rid of all the warnings.

That's why I thought the majority of other rust crates in this org and in rust-vmm have `.idea` in their `.gitignore`.

> https://blog.martinhujer.cz/dont-put-idea-vscode-directories-to-projects-gitignore/

I think this is should not be applied religiously, but on a case-by-case.

# TLDR
I will remove `.idea` if you feel strongly about it, but I think the pros outweigh the cons.

Reply:
Scratch that, I just saw the global `.gitignore` option, that should work nicely.

Reply:
The global gitignore option is quite nice. The problem is more pressing for something like rust-vmm where there are quite a few crates, that's how I discovered the global gitignore as well :))